### PR TITLE
Skip assertion in case of no .tdata

### PR DESCRIPTION
**Pull request by anakrish**
In case of ELF without any .tdata section but with .tbss section,
the linker puts the .tbss offset in the TLS program header
(normally it puts the .tdata offset there).
However the linker still puts the .tdata size (in this case 0)
in the TLS program header.

### CODE REVIEW

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try

File: tests/thread_local/CMakeLists.txt [hunk 1]
```diff
@@ -2,9 +2,10 @@
 # Licensed under the MIT License.
 
 add_subdirectory(host)
-
+add_subdirectory(no_tdata_host)
 if (BUILD_ENCLAVES)
 	add_subdirectory(enc)
+        add_subdirectory(no_tdata_enc)
```
[the remark this answers is not in the document]
Comment (anakrish) [reply tests/thread_local/CMakeLists.txt]: good catch. Fixed the indentation of the entire file.

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try

File: tests/thread_local/no_tdata.edl [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+enclave {
+    trusted {
+        public void enc_set_value(uint64_t value);
```
Comment (CodeMonkeyLeet) [inline tests/thread_local/no_tdata.edl]: Nit: whitespace alignment?

File: tests/thread_local/no_tdata_enc/enc.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
+
```
Comment (CodeMonkeyLeet) [inline tests/thread_local/no_tdata_enc/enc.cpp]: Nit: Remove extra leading whitespace

File: tests/thread_local/no_tdata_host/host.cpp [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        fprintf(stderr, "Usage: %s ENCLAVE_PATH\n", argv[0]);
+        return 1;
+    }
+
+    // Ensure that the enclave has thread-local relocations.
+    elf64_t elf = {0};
+    unsigned char* data = NULL;
+    size_t size = 0;
+
+    // Assert that .tdat seztion does not exist.
```
Comment (CodeMonkeyLeet) [inline tests/thread_local/no_tdata_host/host.cpp]: Typo: ".tdata section"

File: tests/thread_local/no_tdata.edl (hunk 2 above)
Comment (anakrish) [reply tests/thread_local/no_tdata.edl]: Fixed.

File: tests/thread_local/no_tdata_enc/enc.cpp (hunk 3 above)
Comment (anakrish) [reply tests/thread_local/no_tdata_enc/enc.cpp]: Fixed.

File: tests/thread_local/no_tdata_host/host.cpp (hunk 4 above)
Comment (anakrish) [reply tests/thread_local/no_tdata_host/host.cpp]: Fixed.

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try

**Comment by anitagov** [comment]
@anakrish - I see a couple of errors/issues with the last modification of Windows CI script (PR #1321 ) and mentioned them but after the merge. I am fixing both with my ninja Windows CI checkin. 

withEnv(["CC=clang-7","CXX=clang++-7",**"OE_SIMULATION=1"**]) {
                            sh """
                            cmake .. -DCMAKE_BUILD_TYPE=Debug -DUSE_DEBUG_MALLOC=OFF
                            make
                            ctest --output-on-failure
                            """
                        }

The OE_SIMULATION=1 flag should not be passed to cmake and should be removed.
The ctest line is unnecessary too as we don't need to test the ELF enclaves.

I suggest you do one or both of these in your branch.

**Comment by anitagov** [comment]
Updated my previous update. It is JenkinsFile, not Windows PS script

**Pull request merged**